pkg/inject/inject.go

```diff
@@ -170,14 +180,20 @@ func (conf *ResourceConfig) GetPatch(
 		identity := k8s.TLSIdentity{
 			Name:                metaAccessor.GetName(),
 			Kind:                strings.ToLower(conf.meta.Kind),
-			Namespace:           "$" + podNamespaceEnvVarName,
+			Namespace:           "$" + envVarProxyPodNamespace,
 			ControllerNamespace: conf.globalConfig.GetLinkerdNamespace(),
 		}
 
 		report.update(conf)
 		if shouldInject(conf, report) {
-			conf.injectPodSpec(patch, identity)
+			if err := conf.injectPodSpec(patch, identity); err != nil {
+				return nil, nil, err
+			}
 			conf.injectObjectMeta(patch)
+		} else {
+			if err := conf.overrideProxyConfigs(patch); err != nil {
```

Comment:
just want to confirm this behavior for my own understanding. if `shouldInject == false`, but has a `linkerd-proxy` in the pod, `overrideProxyConfigs` will override the existing proxy settings in the pod.

does this mean if i specify some proxy config options during manual injection, those settings will be overridden by auto-inject?

Reply:
> just want to confirm this behavior for my own understanding. if shouldInject == false, but has a linkerd-proxy in the pod, overrideProxyConfigs will override the existing proxy settings in the pod.

That's correct. The current behaviour is that it will always perform an override (even if the old vs. new values are the same), either by using values found in the config annotations, or the defaults from the config map.

> does this mean if i specify some proxy config options during manual injection, those settings will be overridden by auto-inject?

Those config options should be preserved. I still need to confirm with @klingerf on how to preserve them in light of his comment in [2447](https://github.com/linkerd/linkerd2/issues/2447#issuecomment-470320637), if we decided not to store these config options as annotations.